src/ol_infrastructure/infrastructure/consul/__main__.py

```diff
@@ -220,6 +230,49 @@
     opts=ResourceOptions(depends_on=consul_instances),
 )
 
+aws_account = get_caller_identity()
+consul_ami = ec2.get_ami(
+    filters=[
+        ec2.GetAmiFilterArgs(name="name", values=["consul"]),
+        ec2.GetAmiFilterArgs(name="virtualization-type", values=["hvm"]),
+        ec2.GetAmiFilterArgs(name="root-device-type", values=["ebs"]),
+    ],
+    most_recent=True,
+    owners=[aws_account.account_id],
+)
+
+consul_launch_config = ec2.LaunchTemplate(
+    "consul-launch-template",
+    name_prefix=f"consul-{environment_name}-",
+    description="Launch template for deploying Consul cluster",
+    iam_instance_profile=consul_instance_profile.id,
+    image_id=consul_ami.id,
+    instance_type=InstanceTypes[instance_type_name].value,
+    key_name=salt_config.require("key_name"),
+    tags=instance_tags,
```

Comment:
Worth noting that this only applies to the tags on the launch template. There's another parameter `tag_specifications` which handles setting the tags on the instance and volumes.